File: VB/ProcessDuplicatesTarget/ProductReport.vb
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI

Namespace ProcessDuplicatesTarget
	Partial Public Class ProductReport
		Inherits DevExpress.XtraReports.UI.XtraReport

			Public Sub New()
				InitializeComponent()
			End Sub
			' Display duplicate values.
			Public Sub NoMerge()
				Me.ShowPreviewDialog()
			End Sub
			' Merge duplicate values of the XRControl.Tag property.
			Public Sub MergeByTag()
				Dim expressionBinding As New ExpressionBinding("BeforePrint", "Tag", "ToStr([SupplierID]) + '_' + ToStr([CategoryID])")
				Me.xrTableCell2.ExpressionBindings.Add(expressionBinding)
				Me.xrTableCell2.ProcessDuplicatesMode = ProcessDuplicatesMode.Merge
				Me.xrTableCell2.ProcessDuplicatesTarget = DevExpress.XtraReports.UI.ProcessDuplicatesTarget.Tag
				Me.ShowPreviewDialog()
			End Sub
			' Merge duplicate values of a report control's data.
			Public Sub MergeByValue()
				Dim expressionBinding As New ExpressionBinding("BeforePrint", "Text", "[CategoryName]")
				Me.xrTableCell2.ExpressionBindings.Add(expressionBinding)
				Me.xrTableCell2.ProcessDuplicatesMode = ProcessDuplicatesMode.Merge
				Me.xrTableCell2.ProcessDuplicatesTarget = DevExpress.XtraReports.UI.ProcessDuplicatesTarget.Value
				Me.ShowPreviewDialog()
			End Sub
	End Class
End Namespace

File: CS/ProcessDuplicatesTarget/ProductReport.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;

namespace ProcessDuplicatesTarget
{
    public partial class ProductReport : DevExpress.XtraReports.UI.XtraReport
        {
            public ProductReport()
            {
                InitializeComponent();
            }
            // Display duplicate values.
            public void NoMerge()
            {
                this.ShowPreviewDialog();
            }
            // Merge duplicate values of the XRControl.Tag property.
            public void MergeByTag()
            {
                ExpressionBinding expressionBinding = new ExpressionBinding("BeforePrint", "Tag", "ToStr([SupplierID]) + '_' + ToStr([CategoryID])");
                this.xrTableCell2.ExpressionBindings.Add(expressionBinding);
                this.xrTableCell2.ProcessDuplicatesMode = ProcessDuplicatesMode.Merge;
                this.xrTableCell2.ProcessDuplicatesTarget = DevExpress.XtraReports.UI.ProcessDuplicatesTarget.Tag;
                this.ShowPreviewDialog();
            }
            // Merge duplicate values of a report control's data.
            public void MergeByValue()
            {
                ExpressionBinding expressionBinding = new ExpressionBinding("BeforePrint", "Text", "[CategoryName]");
                this.xrTableCell2.ExpressionBindings.Add(expressionBinding);
                this.xrTableCell2.ProcessDuplicatesMode = ProcessDuplicatesMode.Merge;
                this.xrTableCell2.ProcessDuplicatesTarget = DevExpress.XtraReports.UI.ProcessDuplicatesTarget.Value;
                this.ShowPreviewDialog();
            }
        }
}
